framework/src/controller/http/http_server.ts

```diff
@@ -14,43 +14,63 @@
 import * as HTTP from 'http';
 import { Logger } from '../../logger';
 
-export type HTTPRequestListener = (req: HTTP.IncomingMessage, res: HTTP.ServerResponse, message: string) => void;
+export type HTTPRequestListener = (
+	req: HTTP.IncomingMessage,
+	res: HTTP.ServerResponse,
+	message: string,
+) => void;
 
+const ALLOWED_METHODS = ['GET', 'POST'];
 export class HTTPServer {
-  public server!: HTTP.Server;
-  private readonly port: number;
-  private readonly host?: string;
-  private readonly logger: Logger;
-
-  public constructor(options: { port: number; host?: string; logger: Logger }) {
-    this.host = options.host;
-    this.port = options.port;
-    this.logger = options.logger;
-  }
-
-  public start(httpRequestListener: HTTPRequestListener): HTTP.Server {
-    // eslint-disable-next-line @typescript-eslint/no-misused-promises
-    this.server = HTTP.createServer(async (req, res) => {
-      const buffers = [];
-      for await (const chunks of req) {
-        buffers.push(chunks)
-      }
-      const message = Buffer.concat(buffers).toString();
-      return httpRequestListener(req, res, message);
-    });
-    this.server.listen(this.port, this.host, () => {
-      this.logger.info(`HTTP Server is listening at port ${this.port}`)
-    });
-    this.server.on('error', error => {
-      this.logger.error(error);
-    });
-
-    return this.server;
-  }
-
-  public stop(): void {
-    if (this.server) {
-      this.server.close();
-    }
-  }
+	public server!: HTTP.Server;
+	private readonly port: number;
+	private readonly host?: string;
+	private readonly logger: Logger;
+
+	public constructor(options: { port: number; host?: string; logger: Logger }) {
+		this.host = options.host;
+		this.port = options.port;
+		this.logger = options.logger;
+	}
+
+	public start(httpRequestListener: HTTPRequestListener): HTTP.Server {
+		// eslint-disable-next-line @typescript-eslint/no-misused-promises
+		this.server = HTTP.createServer(async (req, res) => {
+			const headers = {
+				'Access-Control-Allow-Origin': '*',
+				'Access-Control-Allow-Methods': ALLOWED_METHODS.join(','),
+				'Content-Type': 'application/json',
+			};
```

Comment:
Can `headers` have an information from `req` or `res` in the future? Otherwise it's a constant piece of information and maybe we can move it out as `ALLOWED_METHODS`.

Reply:
@mehmetegemen this is the server part, i do think we will have anything for request. so the current approach should be good.